test/alternator/test_streams.py

```diff
@@ -0,0 +1,373 @@
+
+def disable_stream(dynamodbstreams, table):
+    print('stooop')
+    while True:
+        try:
+            table.update(StreamSpecification={'StreamEnabled': False});
+            while True:
+                streams = dynamodbstreams.list_streams(TableName=table.name)
+                print(streams)
+                if not streams.get('Streams'):
+                    return
+                print("Stream(s) lingering. Sleep 30s...")
+                time.sleep(30)
+        except ClientError as ce:
+            print("ce {}".format(ce))
+            print("apa {}".format(ce.response['Error']['Code']))
+            if ce.response['Error']['Code'] == 'ResourceInUseException':
+                print("Stream(s) in use. Sleep 30s...")
+                time.sleep(30)
+                continue
+            raise
+            
+#
+# Cannot use fixtures. Because real dynamodb cannot _remove_ a stream
+# once created. It can only expire 24h later. So reusing test_table for 
+# example works great for scylla/local testing, but once we run against
+# actual aws instances, we get lingering streams, and worse: cannot 
+# create new ones to replace it, because we have overlapping types etc. 
+# 
+# So we have to create and delete a table per test. And not run this 
+# test to often against aws.  
+@contextmanager
+def create_stream_test_table(dynamodb, StreamViewType=None):
+    spec = { 'StreamEnabled': False }
+    if StreamViewType != None:
+        spec = {'StreamEnabled': True, 'StreamViewType': StreamViewType}
+    table = create_test_table(dynamodb, StreamSpecification=spec,
+        KeySchema=[ { 'AttributeName': 'p', 'KeyType': 'HASH' },
+                    { 'AttributeName': 'c', 'KeyType': 'RANGE' }
+        ],
+        AttributeDefinitions=[
+                    { 'AttributeName': 'p', 'AttributeType': 'S' },
+                    { 'AttributeName': 'c', 'AttributeType': 'S' },
+        ])
+    yield table
+    while True:
+        try:
+            table.delete()
+            return
+        except ClientError as ce:
+            if ce.response['Error']['Code'] == 'ResourceInUseException':
+                print('Could not delete table yet. Sleeping 5s.')
+                time.sleep(5)
+                continue;
+            raise
+
+def wait_for_active_stream(dynamodbstreams, table, timeout=60):
+    exp = time.process_time() + timeout
+    while time.process_time() < exp:
+        streams = dynamodbstreams.list_streams(TableName=table.name)
+        print(streams)
+        for stream in streams['Streams']:
+            desc = dynamodbstreams.describe_stream(StreamArn=stream['StreamArn'])['StreamDescription']
+            print(desc)
+            if not 'StreamStatus' in desc or desc.get('StreamStatus') == 'ENABLED':
+                arn = stream['StreamArn']
+                if arn != None:
+                    return arn;
+        print("Stream not available. Sleep 5s...")
+        time.sleep(5)
```

Comment:
You will not get 5 seconds, because we need to create new tables most of the time. But again, the loop-delay above only happens when running against real dynamo. Which responds very slowly to stream state changes.

Reply:
Yes, creating tables is a bummer, but even that takes on my laptop about 0.1 seconds (maybe tables with CDC are slower, but probably not significantly slower). What I don't want to see is that table-creation only takes 0.1 seconds but the test already started to sleep for 5 seconds.